### PR TITLE
add video support to segment-anything-2 pipeline

### PR DESCRIPTION
**Pull request by eliteprox**
This change updates the request field `image` to `media_file` and loads the appropriate segment-anything-2 inference model based on the content type of the file. It uses ffmpeg to process the video to image frames and loads them in with inference values from the request. Some adjustments still need to be made to the request/response parameters

### CODE REVIEW

File: runner/app/pipelines/segment_anything_2.py [hunk 1]
```diff
@@ -15,22 +21,73 @@ def __init__(self, model_id: str):
...
-            raise InferenceError(original_exception=e)
+                _, out_obj_ids, out_mask_logits = self.tm_vid.add_new_points_or_box(
+                    inference_state,
+                    frame_idx=5,
+                    obj_id=1,
+                    points=kwargs.get('point_coords', None),
+                    labels=kwargs.get('point_labels', None),
+                    )
+                video_segments = {}  # video_segments contains the per-frame segmentation results
+                for out_frame_idx, out_obj_ids, out_mask_logits in self.tm_vid.propagate_in_video(inference_state):
```
Comment (pschroedl) [inline runner/app/pipelines/segment_anything_2.py:83]: I have a feeling we should return the full triple instead of creating a video segment, leaving post-processing to the consumer of the API ( though I recognize this is good quick way to validate the sanity of the mask outputs )

Comment (eliteprox) [reply runner/app/pipelines/segment_anything_2.py:83]: Thanks for the guidance on this, I see how we can just return the results of `self.tm_vid.propagate_in_video(inference_state):`

Comment (eliteprox) [reply runner/app/pipelines/segment_anything_2.py:96]: > I have a feeling we should return the full triple instead of creating a video segment, leaving post-processing to the consumer of the API ( though I recognize this is good quick way to validate the sanity of the mask outputs )

I had some issues trying to return the correct values. I added frame index as an input parameter, normally `propagate_in_video` will loop returning results for each frame starting at the frame index until the end of the video, now it should only return a single frame. But the data doesn't look correct, can you take a look?
